rtcp-packet.go

```diff
@@ -0,0 +1,360 @@
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+P  |                        report block 2                         :
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+P  |                              ...                              :
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+P  |V=2|P|   SC    |  Packet Type  |              length           |
+   +=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+P  |                          SSRC/CSRC_1                          |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+P  |                           SDES items                          :
+   +=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+P  |                              ...                              :
+   +=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+A  |1|                         SRTCP index                         |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+R  |                  SRTCP MKI (optional) index                   :
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+R  :           SRTCP authentication tag (NOT RECOMMENDED)          :
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+*/
+
+type SRTCPPacket struct {
+  header RTCPHeader
+  payload []byte
+}
+
+func (p *SRTCPPacket) GetESRTCPWord() []byte {
+  len := p.header.GetLengthInBytes() + 12 - RTCPHeaderSize
+  return p.payload[len:len+4]
+}
+
+func (p *SRTCPPacket) GetSRTCPIndex() uint32 {
+  eAndSRTCPIndex := make([]byte, 12)
+  copy(eAndSRTCPIndex, p.GetESRTCPWord())
+
+  // Set the E-bit to 0
+  eAndSRTCPIndex[0] = eAndSRTCPIndex[0] & byte(127)
+
+  return binary.BigEndian.Uint32(eAndSRTCPIndex)
+}
+
+func (p *SRTCPPacket) GetE() bool {
+  return bool((p.GetESRTCPWord()[0] & byte(128)) == 128)
+}
+
+func (p *SRTCPPacket) GetAAD() []byte {
+  srtcpIndexLine := make([]byte, len(p.GetESRTCPWord()))
+  copy(srtcpIndexLine, p.GetESRTCPWord())
+
+  headerBuffer := make([]byte, len(p.header.buffer))
+  copy(headerBuffer, p.header.buffer)
+
+  return append(headerBuffer, srtcpIndexLine...)
+}
+
+func (p *SRTCPPacket) GetCT() []byte {
+  return p.payload[:len(p.payload)-4]
+}
+
+// https://tools.ietf.org/html/rfc7714#section-9.1
+/*
+     0  1  2  3  4  5  6  7  8  9 10 11
+   +--+--+--+--+--+--+--+--+--+--+--+--+
+   |00|00|    SSRC   |00|00|0+SRTCP Idx|---+
+   +--+--+--+--+--+--+--+--+--+--+--+--+   |
+                                           |
+   +--+--+--+--+--+--+--+--+--+--+--+--+   |
+   |         Encryption Salt           |->(+)
+   +--+--+--+--+--+--+--+--+--+--+--+--+   |
+                                           |
+   +--+--+--+--+--+--+--+--+--+--+--+--+   |
+   |       Initialization Vector       |<--+
+   +--+--+--+--+--+--+--+--+--+--+--+--+
+*/
+
+func (p *SRTCPPacket) GgetGCMIV(salt []byte) []byte {
```

Comment:
Typo, but this method should go away once `gcmIV` gets moved onto `RTCPPacket` and `SRTCPPacket` goes away.